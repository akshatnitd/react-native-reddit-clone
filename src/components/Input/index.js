import React from "react"
import PropTypes from "prop-types"
import { Text, TextInput, View, StyleSheet } from "react-native"
import { colorStyle, fontStyle } from "../../style"

const Input = (props) => {
  const { labelText, labelStyle, inputStyle, containerStyle } = props
  return (
    <View style={[style.container, containerStyle]}>
      <Text style={[fontStyle.caption, style.label, labelStyle]}>
        {labelText}
      </Text>
      <TextInput
        style={[
          fontStyle.body1,
          style.input,
          props.multiline ? { paddingBottom: 12 } : null,
          inputStyle,
        ]}
        {...props}
      />
    </View>
  )
}

Input.propTypes = {
  labelText: PropTypes.string,
  labelStyle: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
  inputStyle: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
  containerStyle: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
}

const style = StyleSheet.create({
  container: {
    paddingVertical: 5,
  },
  label: {
    paddingHorizontal: 4,
  },
  input: {
    padding: 4,
    justifyContent: "flex-start",
    textAlignVertical: "top",
  },
})

export default Input
