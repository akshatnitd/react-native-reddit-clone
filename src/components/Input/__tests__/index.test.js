import React from "react"
import renderer from "react-test-renderer"

import Input from ".."

it("renders singleline correctly", () => {
  const tree = renderer.create(<Input labelText="label" />)
  expect(tree.toJSON()).toMatchSnapshot()
})

it("renders multiline correctly", () => {
  const tree = renderer.create(<Input labelText="label" multiline={true} />)
  expect(tree.toJSON()).toMatchSnapshot()
})