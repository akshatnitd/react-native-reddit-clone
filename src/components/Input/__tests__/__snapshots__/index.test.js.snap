// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders multiline correctly 1`] = `
<View
  style={
    Array [
      Object {
        "paddingVertical": 5,
      },
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "color": "rgba(0,0,0,0.54)",
          "fontFamily": "System",
          "fontSize": 12,
          "fontWeight": "400",
          "letterSpacing": 0,
          "lineHeight": 16,
        },
        Object {
          "paddingHorizontal": 4,
        },
        undefined,
      ]
    }
  >
    label
  </Text>
  <TextInput
    allowFontScaling={true}
    labelText="label"
    multiline={true}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "color": "rgba(0,0,0,0.87)",
          "fontFamily": "System",
          "fontSize": 14,
          "fontWeight": "400",
          "letterSpacing": -0.154,
          "lineHeight": 20,
        },
        Object {
          "justifyContent": "flex-start",
          "padding": 4,
          "textAlignVertical": "top",
        },
        Object {
          "paddingBottom": 12,
        },
        undefined,
      ]
    }
  />
</View>
`;

exports[`renders singleline correctly 1`] = `
<View
  style={
    Array [
      Object {
        "paddingVertical": 5,
      },
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "color": "rgba(0,0,0,0.54)",
          "fontFamily": "System",
          "fontSize": 12,
          "fontWeight": "400",
          "letterSpacing": 0,
          "lineHeight": 16,
        },
        Object {
          "paddingHorizontal": 4,
        },
        undefined,
      ]
    }
  >
    label
  </Text>
  <TextInput
    allowFontScaling={true}
    labelText="label"
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "color": "rgba(0,0,0,0.87)",
          "fontFamily": "System",
          "fontSize": 14,
          "fontWeight": "400",
          "letterSpacing": -0.154,
          "lineHeight": 20,
        },
        Object {
          "justifyContent": "flex-start",
          "padding": 4,
          "textAlignVertical": "top",
        },
        null,
        undefined,
      ]
    }
  />
</View>
`;
